httpx/utils.py

```diff
@@ -94,8 +95,9 @@ def guess_json_utf(data: bytes) -> typing.Optional[str]:
 NETRC_STATIC_FILES = (Path("~/.netrc"), Path("~/_netrc"))
 
 
-def get_netrc_login(host: str) -> typing.Optional[typing.Tuple[str, str, str]]:
-    NETRC_FILES = (Path(os.getenv("NETRC", "")),) + NETRC_STATIC_FILES
+@functools.lru_cache()
```

Comment:
Let's limit this with `@functions.lru_cache(1)` so that whenever the environment variable changes we throw away the old one.